Marlin/src/MarlinCore.cpp

```diff
@@ -374,6 +377,9 @@ void disable_all_steppers() {
   DISABLE_AXIS_X();
   DISABLE_AXIS_Y();
   DISABLE_AXIS_Z();
+  #if ENABLED(SOFTWARE_DRIVER_ENABLE)
+    planner.axis_enabled = {false};
+  #endif
```

Comment:
You can probably condense all of these using the `TERN` macros which are used throughout Marlin.
There are a few variations of the `TERN` macros. I believe this is the version you will want to use.

`TERN_(SOFTWARE_DRIVER_ENABLE, planner.axis_enabled.set(false, false, false));`

I believe this will work, although you _might_ need to also use `ENABLED` in there.

`TERN_(ENABLED(SOFTWARE_DRIVER_ENABLE), planner.axis_enabled.set(false, false, false));`

Reply:
Just checked `TERN_` macro, no need to use `ENABLED`.
Moreover, @thinkyhead added more comprehensive TERN macros alternatives two weeks ago, especially [IF_ENABLED](https://github.com/MarlinFirmware/Marlin/blob/0e3d86579bca76e7e47a0fb89064e33514519a67/Marlin/src/core/macros.h#L199).
Will use it.